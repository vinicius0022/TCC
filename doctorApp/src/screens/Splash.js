import React, {Component} from 'react'
import { View, Text, StyleSheet, Image, ImageBackground, Dimensions} from 'react-native';

export default class Splash extends Component {

    componentDidMount = () =>{
      setTimeout(
        () => { this.props.navigation.navigate('Login')},
        2000
      )
    }

    render(){

      return(

        <View style={styles.container}>
          <ImageBackground
            style={styles.rect}
            imageStyle={styles.rect_imageStyle}
            source={require("../../assets/images/Gradient_BlSPiJo.png")}>
          <View style={styles.imageLogo}>
          <Image source={require('../../assets/images/icon.png')} style={styles.image}/>
          <View style={styles.text3Column}>
          <Text style={styles.text3}>Doctor App</Text>
          <View style={styles.rect7}></View>
          </View>
          </View>
          </ImageBackground>
        </View>
      )
    }
}


const styles = StyleSheet.create({

  container:{
    flex: 1,
    alignItems: 'center',
    justifyContent:'center',
    backgroundColor: "rgb(255,255,255)"
  },
  background: {
    flex: 1
  },
  imageLogo:{
    flex: 1,
    justifyContent:'center',
    alignItems:'center',
    flexDirection:'column'
  },
  image:{
    height: 100,
    width:100,
    resizeMode: 'contain',
    justifyContent:'center',
    alignItems:'center'
  },
  header:{
    fontSize:25,
    fontFamily:'shelter'
  },
  rect: {
    flex: 1,
    width: Dimensions.get('window').width
  },
  rect_imageStyle: {
    resizeMode: 'cover'
  },
  text3Column: {
    marginBottom: 6,
    marginLeft: 2,
    marginRight: -1
  },
  text3: {
    color: "rgba(255,255,255,1)",
    fontSize: 25,
    marginBottom: 4
  },
  rect7: {
    height: 8,
    backgroundColor: "white",
    marginRight: 4
  },
})