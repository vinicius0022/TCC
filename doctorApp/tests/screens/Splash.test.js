import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Splash from '../../src/screens/Splash'

describe('Splash', () => {
    it('render', () => {

            //given
            const shallowRenderer = new ShallowRenderer();
            
            //when
            shallowRenderer.render(<Splash/>);
            const result = shallowRenderer.getRenderOutput();

            //Then
            expect(result).toMatchSnapshot();
    });
});