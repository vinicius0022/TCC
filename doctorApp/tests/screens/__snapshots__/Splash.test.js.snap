// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255,255,255)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ImageBackground
    imageStyle={
      Object {
        "resizeMode": "cover",
      }
    }
    source={
      Object {
        "testUri": "../../../assets/images/Gradient_BlSPiJo.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/icon.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 100,
            "justifyContent": "center",
            "resizeMode": "contain",
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "marginBottom": 6,
            "marginLeft": 2,
            "marginRight": -1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(255,255,255,1)",
              "fontSize": 25,
              "marginBottom": 4,
            }
          }
        >
          Doctor App
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "white",
              "height": 8,
              "marginRight": 4,
            }
          }
        />
      </View>
    </View>
  </ImageBackground>
</View>
`;
